src/bus_monitor.cpp

```diff
@@ -10,6 +10,85 @@
 namespace panel
 {
 
+static void resetLEDState()
+{
+    std::cout << "Reset LED state" << std::endl;
+    std::array<const char*, 6> ledsPathOnBasePanel{
+        "/xyz/openbmc_project/led/groups/bmc_booted",
+        "/xyz/openbmc_project/led/groups/power_on",
+        "/xyz/openbmc_project/led/groups/enclosure_identify",
+        "/xyz/openbmc_project/led/groups/platform_system_attention_indicator",
+        "/xyz/openbmc_project/led/groups/partition_system_attention_indicator",
+        "/xyz/openbmc_project/led/groups/enclosure_fault"};
+
+    for (const auto& led : ledsPathOnBasePanel)
+    {
+        auto res = utils::readBusProperty<std::variant<bool>>(
+            "xyz.openbmc_project.LED.GroupManager", led,
+            "xyz.openbmc_project.Led.Group", "Asserted");
+
+        if (auto asserted = std::get_if<bool>(&res))
+        {
+            if (*asserted)
+            {
+                std::cout << "Led asserted = " << led << std::endl;
+                try
+                {
+                    utils::writeBusProperty<bool>(
+                        "xyz.openbmc_project.LED.GroupManager", led,
+                        "xyz.openbmc_project.Led.Group", "Asserted", false);
+
+                    utils::writeBusProperty<bool>(
+                        "xyz.openbmc_project.LED.GroupManager", led,
+                        "xyz.openbmc_project.Led.Group", "Asserted", true);
+                }
+                catch (const sdbusplus::exception::SdBusError& e)
+                {
+                    std::cerr << "Write call for led " << led
+                              << " failed with exception " << e.what()
+                              << std::endl;
+                }
+            }
+        }
+        else
+        {
+            std::cerr
+                << "Failed to read asserted property. Unable to reset led "
+                << led << std::endl;
+        }
+    }
+}
+
+void PanelPresence::readBasePresentProperty(sdbusplus::message::message& msg)
+{
+    if (msg.is_method_error())
+    {
+        std::cerr << "\n Error in reading base panel presence signal "
+                  << std::endl;
+    }
+    std::string interface;
+    types::PropertyValueMap propMap;
+    msg.read(interface, propMap);
+    const auto itr = propMap.find("Present");
+    if (itr != propMap.end())
+    {
+        if (auto present = std::get_if<bool>(&(itr->second)))
+        {
+            transport->setTransportKey(*present);
```

Comment:
Dont set transport key of basae panel as base panel chip resides on bmc card and changing transport key will trigger code update flow.

Reply:
You still have to handle this

Reply:
Yes